src/OrchardCore.Modules/OrchardCore.Queries/Sql/GraphQL/SqlQueryFieldTypeProvider.cs

```diff
@@ -42,18 +45,23 @@ public async Task<IChangeToken> BuildAsync(ISchema schema)
                 var name = query.Name;
                 var source = query.Source;
 
-                var querySchema = JObject.Parse(query.Schema);
-
-                var type = querySchema["type"].ToString();
-
-                if (type.StartsWith("ContentItem/", StringComparison.OrdinalIgnoreCase))
+                try
                 {
-                    var contentType = type.Remove(0, 12);
-                    schema.Query.AddField(BuildContentTypeFieldType(schema, contentType, query));
+                    var querySchema = JObject.Parse(query.Schema);
+                    var type = querySchema["type"].ToString();
+                    if (type.StartsWith("ContentItem/", StringComparison.OrdinalIgnoreCase))
+                    {
+                        var contentType = type.Remove(0, 12);
+                        schema.Query.AddField(BuildContentTypeFieldType(schema, contentType, query));
+                    }
+                    else
+                    {
+                        schema.Query.AddField(BuildSchemaBasedFieldType(schema, query, querySchema));
+                    }
                 }
-                else
+                catch (Exception e)
                 {
-                    schema.Query.AddField(BuildSchemaBasedFieldType(schema, query, querySchema));
+                    _logger.LogError($"The Query {name}'s Schema parse error, Please check the Schema is a correct json format , Schema:{query.Schema}, {System.Environment.NewLine} Error Details:'{e}");
```

Comment:
This is the incorrect logging technique.
It allocates every time when using `$`, but there is a format below, which does not
`_logger.LogError("The query {Name}", name);`

Reply:
ah ，Thanks   . This is a new knowledge point for me !